frontend/server/src/Controllers/User.php

```diff
@@ -3946,6 +3947,48 @@ public static function getIndexDetailsForTypeScript(\OmegaUp\Request $r) {
             \OmegaUp\Controllers\User::ALLOWED_CODER_OF_THE_MONTH_CATEGORIES
         );
 
+        $token = $r->ensureOptionalString(
+            'token',
+            required: false,
+            validator: fn (string $token) => preg_match(
+                '/^[a-zA-Z0-9]{25}$/',
+                $token
+            ) === 1
+        );
+        $parentalVerificationToken = false;
+
+        if (!is_null($token)) {
+            $userId = \OmegaUp\DAO\Users::getUserDataByParentalToken($token);
+
+            if (is_null($userId)) {
+                throw new \OmegaUp\Exceptions\NotFoundException(
+                    'parentalTokenNotFound'
+                );
+            }
+
+            if (is_null($r->user) || is_null($r->user->main_email_id)) {
+                throw new \OmegaUp\Exceptions\UnauthorizedException();
+            }
+
+            $user = \OmegaUp\DAO\Users::getByPK($userId);
+            if (is_null($user)) {
+                throw new \OmegaUp\Exceptions\NotFoundException('userNotExist');
+            }
+
+            try {
+                $user->parent_email_id = $r->user->main_email_id;
+                $user->parent_verified = true;
+                $user->parental_verification_token = null;
+
+                \OmegaUp\DAO\Users::update($user);
+                $parentalVerificationToken = true;
+            } catch (\OmegaUp\Exceptions\ApiException $e) {
+                self::$log->info(
```

Comment:
```suggestion
                self::$log->error(
```

although we shouldn't be catching exceptions in the code! that makes the request succeed even if it shouldn't have. can the `try`/`catch` be removed?